### ISSUE TITLE
API: identifier template issue when column name contains spaces

### ISSUE THREAD

**Issue by martinmaillard** [zazuko/data-cube-curation#194]
```json
{
  "type": "http://tempuri.org/DomainError",
  "title": "Cannot create dimension table",
  "detail": "Template is not correct. Expected \"(\", \"*\", \",\", \":\", \"}\" or [a-zA-Z0-9_.%] but \" \" found.",
  "status": 400
}
```

**Comment by tpluscode** [comment]
Technically API is not really wrong. CSVW uses RFC6570 URI Templates. Spaces are [not allowed in variables](https://tools.ietf.org/html/rfc6570#section-2.3). In fact, only plain ASCII is supported.

Ok, guess the only way is to encode the variables in template. Check https://runkit.com/embed/ykpkxjanhxa3

That would also mean that the rdf-csvw-parser should percent encode column names. I would be surprised if it did that.

**Comment by martinmaillard** [comment]
I will try to encode the variables then.

How should spaces in the rest of the template be handled (e.g. `test with space/{the-column}`? Is this allowed?

**Comment by tpluscode** [comment]
I don't think it makes sense to do any encoding on the UI. Without changes on the API you will get other errors when the names don't match, ie. `'foo bar' != 'foo%20bar'`.

We might as well handle this transparently on the back end completely. 

And might need more change in the parser itself. /cc @bergos

**Comment by martinmaillard** [comment]
Yeah I agree. Just wanted to check what happens... and it seems like it just ignores the variables with spaces.

**Comment by tpluscode** [comment]
Closed by #206
